src/ui/public/field_format_editor/editors/date/date.js

```diff
@@ -0,0 +1,24 @@
+import dateTemplate from './date.html';
+import moment from 'moment';
+
+export function dateEditor() {
+  return {
+    formatId: 'date',
+    editor: {
+      template: dateTemplate,
+      controllerAs: 'cntrl',
+      controller: function ($interval, $scope) {
+        const self = this;
+        self.sampleInputs = [
+          Date.now(),
+          +moment().startOf('year'),
+          +moment().endOf('year')
+        ];
+
+        $scope.$on('$destroy', $interval(function () {
```

Comment:
Yep, you're correct. `$interval` returns a promise, not a function for cancellation. https://code.angularjs.org/1.4.14/docs/api/ng/service/$interval